tests/system/src/test/java/test/com/sun/javafx/text/TextLayoutTest.java

```diff
@@ -0,0 +1,521 @@
+            Font.font("Monaco", 12),
+            0.0f, List.of(180.0293f, 122.41992f),
+            12.0f, 4.001953f
+        )),
+
+        /**
+         * Checks that trailing white space is NOT ignored when wrapping
+         * is not enabled.
+         */
+        HARD_WRAP_WITH_EXTRA_TRAILING_SPACE(new Parameters(
+            "The quick brown fox jumps           \nover the lazy dog           ",
+            Font.font("Monaco", 12),
+            0.0f, List.of(180.0293f + 79.2129f, 122.41992f + 79.2129f),
+            12.0f, 4.001953f
+        )),
+
+        /**
+         * Checks that single white spaces are ignored for alignment
+         * purposes when wrapping is enabled in simple text. This soft
+         * wraps after "jumps".
+         */
+        SOFT_WRAP(new Parameters(
+            "The quick brown fox jumps over the lazy dog",
+            Font.font("Monaco", 12),
+            200.0f, List.of(180.0293f, 122.41992f),
+            12.0f, 4.001953f
+        )),
+
+        /**
+         * Checks that trailing white spaces are NOT stripped even when
+         * some lines are soft wrapped. This soft wraps after "jumps"
+         * and has 4 trailing spaces on the 2nd line where no soft
+         * wrap occurs, and so they should be kept intact.
+         */
+        SOFT_WRAP_WITH_TRAILING_SPACE(new Parameters(
+            "The quick brown fox jumps over the lazy dog    ",
+            Font.font("Monaco", 12),
+            200.0f, List.of(180.0293f, 122.41992f + 28.80469f),
+            12.0f, 4.001953f
+        )),
+
+        /**
+         * Checks that multiple white spaces are ignored for alignment
+         * purposes when wrapping is enabled in simple text. This soft
+         * wraps after "jumps".
+         */
+        SOFT_WRAP_WITH_EXTRA_SPACE(new Parameters(
+            "The quick brown fox jumps           over the lazy dog",
+            Font.font("Monaco", 12),
+            200.0f, List.of(180.0293f, 122.41992f),
+            12.0f, 4.001953f
+        )),
+
+        /**
+         * Checks that single white spaces are ignored for alignment
+         * purposes when wrapping is enabled in complex text. This soft
+         * wraps after "jumps".
+         */
+        SOFT_WRAP_WITH_COMPLEX_TEXT(new Parameters(
+            "The quick brown लोमड़ी jumps over the lazy कुत्ता",
+            Font.font("Monaco", 12),
+            200.0f, List.of(189.89649f, 122.583984f),
+            12.0f, 4.001953f
+        )),
+
+        /**
+         * Checks that multiple white spaces are ignored for alignment
+         * purposes when wrapping is enabled in complex text. This soft
+         * wraps after "jumps".
+         */
+        SOFT_WRAP_WITH_COMPLEX_TEXT_AND_EXTRA_TRAILING_SPACE(new Parameters(
+            "The quick brown लोमड़ी jumps           over the lazy कुत्ता",
+            Font.font("Monaco", 12),
+            200.0f, List.of(189.89649f, 122.583984f),
+            12.0f, 4.001953f
+        ));
+
```

Comment:
Multi line strings are I think addressed in the `hard wrap` cases (they contain line breaks).  I can add a case for leading space (I only have one for trailing space), and can add cases for tabs (although there is uncertainty on how they should be treated).

Reply:
I think we can leave the existing behavior with TABs as is.
I would favor adding a test to capture the existing behavior, since you have the context and are the most knowledgeable person to do so right now :-)